crates/core/src/protocols/fri/prove.rs

```diff
@@ -231,12 +231,10 @@ where
 	message_writer(&mut encoded[..1 << (log_elems - P::LOG_WIDTH)]);
 	rs_code.encode_ext_batch_inplace(&mut encoded, log_batch_size)?;
 
-	// take the first arity as coset_log_len, or use log_inv_rate if arities are empty
-	let coset_log_len = params
-		.fold_arities()
-		.first()
-		.copied()
-		.unwrap_or_else(|| rs_code.log_inv_rate());
+	// Take the first arity as coset_log_len, or use the value such that the number of leaves equals 1 << log_inv_rate if arities is empty
+	let coset_log_len = params.fold_arities().first().copied().unwrap_or_else(|| {
+		log2_strict_usize(encoded.len() * P::WIDTH).saturating_sub(rs_code.log_inv_rate())
+	});
```

Comment:
```suggestion
	let coset_log_len = params.fold_arities().first().copied().unwrap_or_else(|| {
		log_elems
	});
```
this can be simplified.

i think you could also get rid of the other `log2_strict_usize` below, but it's slightly less obvious how

Reply:
oh, thank you!